Automattic/wp-calypso
# Fix redirect to WooCommerce home WooCommerce related plans

Related to https://github.com/Automattic/wc-calypso-bridge/issues/1214

## Proposed Changes

* Adds `isSiteOnWooExpressEcommerceTrial` plan selector to detect Woo Express, Ecommerce, free trial plans
* Fix redirect to WooCommerce home for all mentioned plans

## Testing Instructions

### Positive test

1. Using a site with a free trial plan, go to `https://calypso.localhost:3000/home/site_url`
2. Observe that you're redirected to WooCommerce homescreen
3. Repeat the test using a site with Woo Express medium or small plan, and Ecommerce plan

### Negative test

1. Using a site without Woo Express or Ecommerce plan, go to `https://calypso.localhost:3000/home/site_url`
2. Observe that you're not redirected to anywhere else

### Test with SSO disabled

- Navigate to **Settings > Security** and disable the "_Allow users to log in to this site using WordPress.com accounts_" option.
- Navigate to the site's home (`wordpress.com/home/[site_suffix]`) that has SSO disabled.
- Notice that you get redirected to WooCommerce Home (or the wp-admin login page, depending on the login cookie).
- Apply this PR.
- Navigate to home again and ensure that the redirection is removed; therefore, you should see the Calypso home.

## Pre-merge Checklist
